### PR TITLE
flutter analyze command

### PR DESCRIPTION
**Pull request by Hixie**
Other changes in this patch:
- Make the 'flutter' tool say "Updating flutter tool..." when it calls
  pub get, to avoid confusion about what the pub get output is about.
- Make the bash flutter tool call pub get when the revision has
  changed. (This was already happening on Windows.)
- Fix a raft of bugs found by the analyzer.
- Fix some style nits in various bits of code that happened to be near
  things the analyzer noticed.
- Remove the logic in "flutter test" that would run "pub get", since
  upon further reflexion it was determined it didn't work anyway.
  We'll probably have to add better diagnostics here and say to run the
  updater script.
- Remove the native velocity tracker script, since it was testing code
  that has since been removed.

Notes on ignored warnings:
- We ignore warnings in any packages that are not in the Flutter repo or
  in the author's current directory.
- We ignore various irrelevant Strong Mode warnings. We still enable
  strong mode because even though it's not really relevant to our needs,
  it does (more or less accidentally) catch a few things that are
  helpful to us.
- We allow CONSTANTS_LIKE_THIS, since we get some of those from other
  platforms that we are copying for sanity and consistency.
- We allow one-member abstract classes since we have a number of them
  where it's perfectly reasonable.
- We unfortunately still ignore warnings in mojom.dart autogenerated
  files. We should really fix those but that's a separate patch.
- We verify the actual source file when we see the 'Name non-constant
  identifiers using lowerCamelCase.' lint, to allow one-letter variables
  that use capital letters (e.g. for physics expressions) and to allow
  multiple-underscore variable names.
- We ignore all errors on lines that contain the following magic
  incantation and a "#" character:
    // analyzer doesn't like constructor tear-offs
- For all remaining errors, if the line contains a comment of the form
  // analyzer says "..."
  ...then we ignore any errors that have that "..." string in them.

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/analyze.dart [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+import 'package:logging/logging.dart';
+import 'package:path/path.dart' as path;
+
+import '../artifacts.dart';
+import '../build_configuration.dart';
+import '../process.dart';
+import 'flutter_command.dart';
+
+final Logger _logging = new Logger('sky_tools.test');
```
Comment (abarth) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: sky_tools.test -> sky_tools.analyze

File: packages/flutter_tools/lib/src/commands/analyze.dart [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+    argParser.addFlag('flutter-repo', help: 'Include all the examples and tests from the Flutter repository.', defaultsTo: false);
+    argParser.addFlag('current-directory', help: 'Include all the Dart files in the current directory, if any.', defaultsTo: true);
+    argParser.addFlag('current-package', help: 'Include the lib/main.dart file from the current directory, if any.', defaultsTo: true);
+    argParser.addFlag('congratulate', help: 'Show output even when there are no errors, warnings, hints, or lints.', defaultsTo: true);
+  }
+
+  bool get requiresProjectRoot => false;
+
+  @override
+  Future<int> runInProject() async {
```
Comment (abarth) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: Should this just be "run" ?

**Comment by abarth** [comment]
LGTM

File: packages/flutter_tools/lib/src/commands/analyze.dart (hunk 2 above)
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: https://github.com/flutter/flutter/issues/333

**Pull request merged**